packages/hardhat/test/ChainLance.ts

```diff
@@ -13,15 +13,240 @@ describe("ChainLance", function () {
+    const txAcceptWork = await chainLance.connect(employer).acceptWork(projectId);
+    const receiptAcceptWork = await txAcceptWork.wait();
+    expect(receiptAcceptWork?.status).is.equal(1, "Work acception failed");
+    if (receiptAcceptWork) {
+      expect(await ethers.provider.getBalance(worker.address)).is.greaterThanOrEqual(
+        BigInt(projectPrice) - receiptAcceptWork.gasPrice * receiptAcceptWork.gasUsed,
+        "Worker did not recieve payment",
+      );
+    }
+
+    // Worker has 0 balance edge case accounted with gtThanOrEq
+  });
+
+  it("Should allow cancelling/rejecting during the work on the project", async function () {
+    const [employer, worker] = await ethers.getSigners();
+
+    const projectId = "id2";
+    const projectPrice = 200000;
+    const timeSpanToCancel = 100;
+    const txCreateProject = await chainLance.connect(employer).createProject(projectId, projectPrice, timeSpanToCancel);
+    const receiptCreateProject = await txCreateProject.wait();
+
+    expect(receiptCreateProject).is.not.null;
+    expect(receiptCreateProject?.status).is.equal(1, "project creation failed");
+
+    const bidId = "bid2";
+    await chainLance.connect(worker).bidProject(projectId, bidId, projectPrice, timeSpanToCancel - 1); // timespan on project is for bidding?
+
+    await chainLance.connect(employer).acceptBid(projectId, bidId, { value: projectPrice });
+
+    await chainLance.connect(worker).submitWork(projectId);
+
+    ///////////////////////////////////////////////////////////////////
+    // Reject work submition
+    ///////////////////////////////////////////////////////////////////
+    const txRejectWork = await chainLance.connect(employer).rejectWork(projectId);
+    const receiptRejectWork = await txRejectWork.wait();
+    expect(receiptRejectWork?.status).is.equal(1, "work rejection failed");
+
+    ///////////////////////////////////////////////////////////////////
+    // Cancel work
+    ///////////////////////////////////////////////////////////////////
+
+    await ethers.provider.send("evm_increaseTime", [timeSpanToCancel + 1]); // Changing block timestamp
+    await ethers.provider.send("evm_mine", []); // Mine block
+    const txCancelWork = await chainLance.connect(employer).cancelWork(projectId);
+    const receiptCancelWork = await txCancelWork.wait();
+    expect(receiptCancelWork?.status).is.equal(1, "work cancellation failed");
+  });
+});
+
+describe("ChainLance: query info on project", function () {
+  // We define a fixture to reuse the same setup in every test.
+
+  let chainLance: ChainLance;
+  before(async () => {
+    // const [owner] = await ethers.getSigners();
+    const chainLanceFactory = await ethers.getContractFactory("ChainLance");
+    chainLance = (await chainLanceFactory.deploy()) as ChainLance;
+    await chainLance.waitForDeployment();
+
+    const [employer1, employer2, employer3, worker1, worker2, worker3] = await ethers.getSigners(); // can i do that?
```

Comment:
Yes. REmove this comment. You can leave comments like this one in PR draft instead.